### PR TITLE
SM-1163 / Handle carriage returns

### PR DESCRIPTION
**Pull request by mmdotz**
Bug in Excel for Mac saves csvs with \r carriage returns

This removes the carriage returns from the string. Tested locally outside of SA.

### CODE REVIEW

File: lib/eightball.rb [hunk 1]
```diff
@@ -1,12 +1,11 @@
 # encoding: utf-8
 
 module Eightball
-
   def eightball(replacement_string = '?')
-    string = self
```
Comment (cath-oneill) [inline lib/eightball.rb]: Why did this line get removed?  I'd be concerned that things don't work with this line removed.

File: test/test_eightball.rb [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
+    puts new_string.eightball
+  end
+
+  should 'remove carriage return in test' do
+    string = File.read('./files/test_with_returns1.csv')
+    string = string.eightball
+    assert !string.empty?
+    assert string.valid_encoding?
+    assert string.encoding == Encoding::UTF_8
+  end
```
Comment (cath-oneill) [inline test/test_eightball.rb]: So excited that you wrote tests!!!!!!

File: test/test_eightball.rb [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
     assert string.valid_encoding?
     assert string.encoding == Encoding::UTF_8
   end
 
+  should 'remove carriage return if exists' do
+    test_string = "foobarbarbaz\r\n"
+    new_string = "foobarbarbaz\n"
+    assert_equal new_string, test_string.eightball
+    assert !new_string.include?("\r")
+    puts new_string.eightball
```
Comment (cath-oneill) [inline test/test_eightball.rb]: Do we need this puts statement?

**Review by cath-oneill — APPROVED** [review]
This looks good!  I had a couple small comments/questions, but nothing big.  And I'm super happy that you wrote tests!!!!

File: test/test_eightball.rb (hunk 3 above)
Comment (mmdotz) [reply test/test_eightball.rb]: Not anymore!

File: lib/eightball.rb (hunk 1 above)
Comment (mmdotz) [reply lib/eightball.rb]: Rubocop suggested it was redundant

**Pull request merged**